service/controller/clusterapi/v20/resources/configmap/desired.go

```diff
@@ -116,6 +114,37 @@ func (r *Resource) newChartSpecs() []pkgkey.ChartSpec {
 	}
 }
 
+func (r *Resource) newIngressControllerValues() (IngressControllerValues, error) {
+	switch r.provider {
+	case "aws":
+		return IngressControllerValues{
+			// Controller service is disabled because manifest is created by
+			// Ignition.
+			ControllerServiceEnabled: false,
+			// Proxy protocol is enabled for AWS clusters.
+			UseProxyProtocol: true,
+		}, nil
+	case "azure":
+		return IngressControllerValues{
+			// Controller service is disabled because manifest is not created by
+			// Ignition.
+			ControllerServiceEnabled: true,
```

Comment:
On Azure we didn't need to migrate existing clusters. So the IC service is created by the chart.
https://github.com/giantswarm/kubernetes-nginx-ingress-controller/blob/f91ca672a7659a0264359922c3809be0d7d27e8a/helm/kubernetes-nginx-ingress-controller-chart/templates/controller-service.yaml#L1

Maybe we can do the same for AWS for Cluster API clusters. For now, this just gets the logic straight.

cc @puja108 @corest @pipo02mix Since we discussed this earlier.